airbyte-webapp/src/packages/cloud/components/experiments/FreeConnectorProgram/EnrollmentModal/useShowEnrollmentModal.tsx

```diff
@@ -10,6 +15,18 @@ export const useShowEnrollmentModal = () => {
   const { mutateAsync: createCheckout } = useStripeCheckout();
   const workspaceId = useCurrentWorkspaceId();
   const { emailVerified, sendEmailVerification } = useAuthService();
+  const { formatMessage } = useIntl();
+  const { registerNotification } = useNotificationService();
+
+  const verifyEmail = () =>
+    sendEmailVerification().then(() => {
+      registerNotification({
+        id: "fcp/verify-email",
+        text: formatMessage({ id: "freeConnectorProgram.enrollmentModal.validationEmailConfirmation" }),
+        type: ToastType.INFO,
+      });
+      closeModal();
```

Comment:
> I think closing the modal is kind of weird UX

Agreed. Unfortunately, because the modal is at a higher z-index than the toast, leaving it open means the only way to confirm the action is rendering something inline in the modal, which also seemed weird.

Reply:
I think we can switch those z-indexes around. In general I'd assume an incoming notification should be higher even than an open modal?

Reply:
And for posterity (talked offline with Joey about that): We should be fine reordering the providers, but we might need to check which orders work since there are "some" dependencies between them.

Reply:
:+1: to both. To keep this particular PR's diff focused, though, I'll leave the providers reastructuring and clean up associated cleanup for a follow-up PR.